### PR TITLE
Docs: Doc Blocks  updates for known limitations

### LINKED ISSUE #22267: Clarify limitations of `<Story inline={false} />`
It breaks controls.

Need to update API references for both [Story](https://storybook.js.org/docs/api/doc-block-story) and [Controls](https://storybook.js.org/docs/api/doc-block-controls)

See: https://discord.com/channels/486522875931656193/1087584071573577798/1087584071573577798

### LINKED ISSUE #24267: [Documentation]: excludeDecorators property missing in docs
### Describe the problem

The `excludeDecorators` not documented on the source doc block pages, e.g.:
- https://storybook.js.org/docs/web-components/api/doc-block-source
- https://storybook.js.org/docs/web-components/api/doc-block-source
- https://storybook.js.org/docs/vue/api/doc-block-source
- https://storybook.js.org/docs/angular/api/doc-block-source

Property source:
https://github.com/storybookjs/storybook/blob/a12a65bb2ae79b4b0efb36131a3a6c7faa74bc31/code/renderers/html/src/types.ts#L34

The property was added with this issue fix:
https://github.com/storybookjs/storybook/issues/12022#issuecomment-859540813

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jonniebigodes**
Closes #24267  and closes #22267

## What I did

With this small pull request the relevant sections were updated to reflect some of the gaps and limitations of generating documentation with Storybook.

@JReinhold, when you have a moment, can you take a look and let me know of any feedback you may have so that we can close out the associated issues? Thanks in advance.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: docs/api/doc-block-controls.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ import * as ButtonStories from './Button.stories'
 
 <Callout variant="warning">
 
-The Controls doc block will only have functioning UI controls if you have also installed and registered [`@storybook/addon-controls`](../essentials/controls.md) (included in [`@storybook/addon-essentials`](../essentials/index.md)).
+The Controls doc block will only have functioning UI controls if you have also installed and registered [`@storybook/addon-controls`](../essentials/controls.md) (included in [`@storybook/addon-essentials`](../essentials/index.md)) and haven't turned off inline stories with the [`inlineStories`](./doc-block-story.md#inline) configuration option.
```
Comment (kylegach) [inline docs/api/doc-block-controls.md:34]: It's confusing to reference a non-existent property with the code formatting. The preceding "inline stories" adds enough clarity, I think.

```suggestion
The Controls doc block will only have functioning UI controls if you have also installed and registered [`@storybook/addon-controls`](../essentials/controls.md) (included in [`@storybook/addon-essentials`](../essentials/index.md)) and haven't turned off inline stories with the [`inline`](./doc-block-story.md#inline) configuration option.
```

File: docs/api/doc-block-source.md [hunk 2]
```diff
@@ -98,6 +98,18 @@ Light mode is only supported when the `Source` block is rendered independently.
 
 </Callout>
 
+<IfRenderer renderer={['angular', 'react', 'html', 'web-components' ]}>
+
+### `excludeDecorators`
+
+Type:
```
Comment (kylegach) [inline docs/api/doc-block-source.md:105]:
```suggestion
Type: `boolean`
```

File: docs/writing-docs/autodocs.md [hunk 3]
```diff
@@ -286,6 +286,10 @@ Additionally, if you're developing using TypeScript, you may need to update Stor
 
 If you're still encountering issues, we recommend reaching out to the community using the default communication channels (e.g., [GitHub discussions](https://github.com/storybookjs/storybook/discussions/new?category=help)).
 
+### The controls are not updating the story within the auto-generated documentation
+
+If you turned off inline rendering for your stories via the [`inlineStories`](../api/doc-block-story.md) configuration option, you would run into a situation where the associated controls are not updating the story within the documentation page. This is a known limitation of the current implementation and will be addressed in a future release.
```
Comment (kylegach) [inline docs/writing-docs/autodocs.md:291]:
```suggestion
If you turned off inline rendering for your stories via the [`inline`](../api/doc-block-story.md) configuration option, you would run into a situation where the associated controls are not updating the story within the documentation page. This is a known limitation of the current implementation and will be addressed in a future release.
```

File: docs/writing-docs/mdx.md [hunk 4]
```diff
@@ -455,6 +455,10 @@ Additionally, if you're working with VSCode, you can add the [MDX extension](htt
 
 If you're still encountering issues, we recommend reaching out to the community using the default communication channels (e.g., [GitHub discussions](https://github.com/storybookjs/storybook/discussions/new?category=help)).
 
+### The controls are not updating the story within the MDX documentation page
+
+If you turned off inline rendering for your stories via the [`inlineStories`](../api/doc-block-story.md) configuration option, you would run into a situation where the associated controls are not updating the story within the documentation page. This is a known limitation of the current implementation and will be addressed in a future release.
```
Comment (kylegach) [inline docs/writing-docs/mdx.md:460]:
```suggestion
If you turned off inline rendering for your stories via the [`inline`](../api/doc-block-story.md) configuration option, you would run into a situation where the associated controls are not updating the story within the documentation page. This is a known limitation of the current implementation and will be addressed in a future release.
```

**Review by kylegach — APPROVED** [review]
Good updates! Made a few suggestions, but they're easy to incorporate, so I'm going to go ahead and approve to un-block you.

**Review by JReinhold — APPROVED** [review]

**Pull request merged**